packages/phase/test/unit/phase.unit.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright IBM Corp. 2014. All Rights Reserved.
+// Node module: loopback-phase
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as assert from 'assert';
+import {expect} from '@loopback/testlab';
+import {Phase} from '../..';
+import {Handler} from '../../src';
+
+describe('Phase', () => {
+  describe('phase.run(ctx)', () => {
+    it('should execute phase handlers', async () => {
+      const phase = new Phase();
+      let called: boolean | undefined;
+      phase.use(async ctx => {
+        called = true;
+      });
+      await phase.run();
+      assert(called === true);
+    });
+
+    it('should set the context for handlers', async () => {
+      const phase = new Phase();
+      phase.use(async ctx => {
+        expect(ctx).to.have.property('foo', 'bar');
+      });
+      await phase.run({foo: 'bar'});
+    });
+
+    describe('execution order', () => {
+      let called: string[];
+      let mockHandler: (name: string) => Handler;
+
+      beforeEach(() => {
+        called = [];
+        mockHandler = function(name: string) {
+          return ctx => {
+            called.push(name);
+            return new Promise<void>(resolve => {
+              process.nextTick(() => {
+                called.push(name + '_done');
+                resolve();
+              });
+            });
+          };
+        };
+      });
+
+      it('should execute phase handlers in parallel', async () => {
+        const phase = new Phase({parallel: true});
+
+        phase
+          .before(mockHandler('b1'))
+          .before(mockHandler('b2'))
+          .use(mockHandler('h1'))
+          .after(mockHandler('a1'))
+          .after(mockHandler('a2'))
+          .use(mockHandler('h2'));
```

Comment:
Please use a more descriptive names, I find it difficult to follow these terse two-letter identifiers.

For example: `before-handlerA-1`, `before-handlerA-2`, `handlerA`, `after-handlerA-1`, etc. `handlerB`.